### PR TITLE
Add hcp-density-aks workload

### PR DESCRIPTION
**Pull request by krishvoor**
This PR adds a new workload hcp-density-aks.
Few updates the aks-arm-template.json as well in the process

### CODE REVIEW

File: kube-burner/hcp-density-aks/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        node-role.kubernetes.io/worker: ""
+      containers:
+      - args:
+        - sleep
+        - infinity
+        image: registry.k8s.io/pause:3.1
+        resources:
+          requests:
+            cpu: 10m
+            memory: 10M
```
Comment (krishvoor) [inline kube-burner/hcp-density-aks/templates/deployment.yaml]: These requests have to be changed according to [Grafana URL](https://grafana.rdu2.scalelab.redhat.com:3000/d/SNoAlkE4k/hypershift-report-multi-hc?orgId=1&from=1706890903943&to=1706895895604&var-Datasource=AWS+Pro+-+ripsaw-kube-burner&var-mgmt_cluster_name=aromc1-djt22&var-workload=cluster-density-v2&var-svc_cluster_name=&var-latencyPercentile=P99&var-nonServingWorker=All&var-servingWorker=All&var-uuid=80f4401f-4be0-49e3-9757-d6db4b7968ec&var-hosted_cluster_name=aro-hosted-cp-7wncl)

File: kube-burner/hcp-density-aks/hcp-density-aks.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  indexerConfig:
+    esServers: [http://elastic-elk.apps.rsevilla-ocp45-ovn.perf-testing.devcluster.openshift.com]
+    insecureSkipVerify: false
+    defaultIndex: kube-burner
+    type: elastic
+  measurements:
+    - name: podLatency
+  clusterHealth: true
+
+jobs:
```
Comment (mukrishn) [inline kube-burner/hcp-density-aks/hcp-density-aks.yml:12]: config maps & secrets ?

Comment (krishvoor) [reply kube-burner/hcp-density-aks/hcp-density-aks.yml:12]: Did not find any secrets/configmaps here :(
https://gist.github.com/mukrishn/05964efa55cde274067375aa1bfd2ef8

Comment (mukrishn) [reply kube-burner/hcp-density-aks/hcp-density-aks.yml:12]: my bad, `get all` didn't capture it. let me update it

Comment (mukrishn) [reply kube-burner/hcp-density-aks/hcp-density-aks.yml:12]: updated it, there are approxly 110 secrets. keep the same count for configmap too.

File: kube-burner/hcp-density-aks/hcp-density-aks.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+      - objectTemplate: templates/deployment.yaml
+        replicas: 45
+        inputVars:
+          name: deployment-3pods
+
+      - objectTemplate: templates/deployment.yaml
+        replicas: 6
+        inputVars:
+          name: deployment-pod
```
Comment (mukrishn) [inline kube-burner/hcp-density-aks/hcp-density-aks.yml:45]: can we also add a few multi-container pod ?

Comment (krishvoor) [reply kube-burner/hcp-density-aks/hcp-density-aks.yml:45]: done, ptal

**Review by mukrishn — APPROVED** [review]

**Pull request merged**